vignettes/agg_scale_performance.Rmd

```diff
@@ -0,0 +1,175 @@
+---
+title: "Aggregation/Scaling performance"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Aggregation/Scaling performance}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+  
+Aggregations and scaling with `hierarchyUtils` should perform as fast as basic data.table code.
+For basic use cases there should only be slightly more overhead due to the assertions and added flexibility included in `hierarchyUtils`.
+
+```{r, include=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  echo = FALSE,
+  message = FALSE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(hierarchyUtils)
+library(data.table)
+
+profile_n_draws <- c(1, 10)
+```
+
+```{r time_aggregation_function}
+time_aggregation <- function(n_draws,
```

Comment:
Include a short description for each of these function parameters? Or a note that most of them come from `hierarchyUtils::agg`?